<a href="https://colab.research.google.com/github/AsmaBenHmidene/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Create a notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

First choice: dataset 1
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Second choice: dataset 2
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Create a NEW GitHub repository for this project, with a readme, and add your notebook to that new repository. Submit the link below.

# First choice: dataset 1
## Metabolic Syndrome:

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
import pandas as pd
pd.set_option('display.max_columns',100)

In [4]:
# Load data from file structure
filename = '/content/drive/MyDrive/Coding Dojo/Project 2/Metabolic  Syndrome.csv'
df = pd.read_csv(filename)
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

1. Source of data

The dataset for analysis came from the NHANES initiative.
https://data.world/informatics-edu/metabolic-syndrome-prediction

2. Brief description of data

To predict metabolic syndrome, yes or not based on common risk factors.The following variables were combined from multiple tables with SQL: abnormal waist circumference, triglycerides above 150, HDL cholesterol below 50 in women or 40 in men, history of hypertension and mildly elevated fasting blood sugar (100-125). Numerous other variables were added, such as uric acid, race, income, etc. that might contribute to the model

3. What is the target?

Metabolicsyndrome

4. What does one row represent? (A person? A business? An event? A product?)

a person

5. Is this a classification or regression problem?

classification problem

6. How many features does the data have?

14 features

7. How many rows are in the dataset?

2401 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Some features might not be relevant to  the model.

# Second choice: dataset 2
## Thyroid:

In [5]:
# Load data from file structure
filename2 = '/content/drive/MyDrive/Coding Dojo/Project 2/thyroidDF.csv'
df2 = pd.read_csv(filename2)
df2.info()
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9172 entries, 0 to 9171
Data columns (total 31 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   age                  9172 non-null   int64  
 1   sex                  8865 non-null   object 
 2   on_thyroxine         9172 non-null   object 
 3   query_on_thyroxine   9172 non-null   object 
 4   on_antithyroid_meds  9172 non-null   object 
 5   sick                 9172 non-null   object 
 6   pregnant             9172 non-null   object 
 7   thyroid_surgery      9172 non-null   object 
 8   I131_treatment       9172 non-null   object 
 9   query_hypothyroid    9172 non-null   object 
 10  query_hyperthyroid   9172 non-null   object 
 11  lithium              9172 non-null   object 
 12  goitre               9172 non-null   object 
 13  tumor                9172 non-null   object 
 14  hypopituitary        9172 non-null   object 
 15  psych                9172 non-null   o

age sex on_thyroxine query_on_thyroxine on_antithyroid_meds sick pregnant  \
0   29   F            f                  f                   f    f        f   
1   29   F            f                  f                   f    f        f   
2   41   F            f                  f                   f    f        f   
3   36   F            f                  f                   f    f        f   
4   32   F            f                  f                   f    f        f   

  thyroid_surgery I131_treatment query_hypothyroid query_hyperthyroid lithium  \
0               f              f                 t                  f       f   
1               f              f                 f                  f       f   
2               f              f                 f                  t       f   
3               f              f                 f                  f       f   
4               f              f                 f                  f       f   

  goitre tumor hypopituitary psych TSH_measured  TSH T3_measured   T3  \
0      f     f             f     f            t  0.3           f  NaN   
1      f     f             f     f            t  1.6           t  1.9   
2      f     f             f     f            f  NaN           f  NaN   
3      f     f             f     f            f  NaN           f  NaN   
4      f     f             f     f            f  NaN           f  NaN   

  TT4_measured    TT4 T4U_measured  T4U FTI_measured  FTI TBG_measured   TBG  \
0            f    NaN            f  NaN            f  NaN            f   NaN   
1            t  128.0            f  NaN            f  NaN            f   NaN   
2            f    NaN            f  NaN            f  NaN            t  11.0   
3            f    NaN            f  NaN            f  NaN            t  26.0   
4            f    NaN            f  NaN            f  NaN            t  36.0   

  referral_source target  patient_id  
0           other      -   840801013  
1           other      -   840801014  
2           other      -   840801042  
3           other      -   840803046  
4           other      S   840803047

1. Source of data

Thyroid Data - https://archive.ics.uci.edu/ml/datasets/thyroid+disease
https://www.kaggle.com/datasets/emmanuelfwerr/thyroid-disease-data

2. Brief description of data

The datasets featured were created by reconciling thyroid disease datasets provided by the UCI Machine Learning Repository of patient background information and blood test results along with thyroid-related medical diagnosis

3. What is the target?

target - hyperthyroidism medical diagnosis (str).

4. What does one row represent? (A person? A business? An event? A product?)

a person

5. Is this a classification or regression problem?

classification problem.

6. How many features does the data have?

30 features.

7. How many rows are in the dataset?

9172 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

many missing values in multiple columns.